GooglePlayInstant/LoadingScreen/LoadingBar.cs

```diff
@@ -0,0 +1,184 @@
+        private const string LoadingBarGameObjectName = "Loading Bar";
+        private const string LoadingBarOutlineGameObjectName = "Loading Bar Outline";
+        private const string LoadingBarFillGameObjectName = "Loading Bar Fill";
+
+#if UNITY_EDITOR
+        /// <summary>
+        /// Creates a loading bar component on the specified loading screen game object's bottom half. Consists of
+        /// a white rounded border, with a colored loading bar fill in the middle.
+        /// </summary>
+        public static void AddComponent(GameObject loadingScreenGameObject)
+        {
+            var loadingBarGameObject = new GameObject(LoadingBarGameObjectName);
+            loadingBarGameObject.AddComponent<RectTransform>();
+            loadingBarGameObject.transform.SetParent(loadingScreenGameObject.transform);
+
+            SetOutline(loadingBarGameObject);
+            SetFill(loadingBarGameObject);
+        }
+
+        // TODO: check for compatibilty with unity 5.6+
+        private static void SetOutline(GameObject loadingBarGameObject)
+        {
+            var loadingBarOutlineGameObject = new GameObject(LoadingBarOutlineGameObjectName);
+            loadingBarOutlineGameObject.transform.SetParent(loadingBarGameObject.transform);
+
+            loadingBarOutlineGameObject.AddComponent<Image>();
+
+            var loadingBarOutlineImage = loadingBarOutlineGameObject.GetComponent<Image>();
+            loadingBarOutlineImage.sprite =
+                AssetDatabase.GetBuiltinExtraResource<Sprite>("UI/Skin/InputFieldBackground.psd");
+
+            loadingBarOutlineImage.type = Image.Type.Sliced;
+            loadingBarOutlineImage.fillCenter = false;
+
+            // Position outline component
+            loadingBarOutlineGameObject.transform.position = loadingBarGameObject.transform.position;
+        }
+
+        private static void SetFill(GameObject loadingBarGameObject)
+        {
+            var loadingBarFillGameObject = new GameObject(LoadingBarFillGameObjectName);
+            loadingBarFillGameObject.transform.SetParent(loadingBarGameObject.transform);
+
+            loadingBarFillGameObject.AddComponent<Image>();
+            var loadingBarFillImage = loadingBarFillGameObject.GetComponent<Image>();
+            loadingBarFillImage.color = Color.green;
+
+            // Position fill component
+            loadingBarFillGameObject.transform.position = loadingBarGameObject.transform.position;
+        }
+#endif
+
+        /// <summary>
+        /// Sets the position and size of the loading bar as a function of the screen heighta and width. 
+        /// </summary>
+        public static void UpdateSizeAndPostition()
+        {
+            var loadingBarRectTransform = GameObject.Find(LoadingBarGameObjectName).GetComponent<RectTransform>();
+            var loadingBarFillRectTransform =
+                GameObject.Find(LoadingBarFillGameObjectName).GetComponent<RectTransform>();
```

Comment:
If we can, we should store a reference to the loading bar objects instead of doing 'GameObject.Find()' and 'GetComponent' every frame. It's likely not a performance issue in this case because we expect the Scene to not have very many objects in it. However, it's good form to use those functions sparingly.

Reply:
Those references could be stored in that LoadingBar MonoBehaviour in my comment above.

Reply:
I think you're both making good points since "Update" often means something special in Unity:
https://docs.unity3d.com/ScriptReference/MonoBehaviour.Update.html

We should rename.